### PR TITLE
As a user, I can edit a link

### PR DESCRIPTION
**Pull request by suho**
- Endpoint: /links/:id
- Method: PATCH
- Parameter:
```json
{
    "alias": "facebook",
    "password": "123456"
}
```

- Response:
```json
{
    "id": 6,
    "created_at": "2020-08-11T12:01:08.173106+07:00",
    "updated_at": "2020-08-11T12:01:08.173106+07:00",
    "deleted_at": null,
    "original_url": "https://www.facebook.com",
    "alias": "facebook",
    "password": "123456"
}
```

### CODE REVIEW

File: controllers/link.go [hunk 1]
```diff
@@ -146,6 +151,55 @@ func GetLinkWithPassword(c *gin.Context) {
...
 }
 
+func UpdateLink(c *gin.Context) {
+	var input UpdateLinkInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.JSON(http.StatusUnprocessableEntity, err.Error())
+		return
+	}
+
+	if input.Password == "" && input.Alias == "" {
```
Comment (namtx) [inline controllers/link.go:161]: how about add `binding:required` to `UpdateLinkInput` struct on two of these fields?
what do you think?

Comment (suho) [reply controllers/link.go:161]: Currently, we just want to required at least one field, can we do that with `binding:required`?

File: controllers/link.go [hunk 2]
```diff
@@ -146,6 +151,55 @@ func GetLinkWithPassword(c *gin.Context) {
...
+		return
+	}
+
+	if input.Password != "" {
+		link.Password = input.Password
+	}
+
+	if input.Alias != "" {
+		if linkutils.IsDuplicateAlias(input.Alias) {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Duplicate Alias"})
```
Comment (namtx) [inline controllers/link.go]:
```suggestion
			c.JSON(http.StatusBadRequest, gin.H{"error": "This alias  is already used"})
```
Would be more informative, what do you think?

File: controllers/link.go (hunk 1 above)
Comment (namtx) [reply controllers/link.go:161]: got it

File: controllers/link.go [hunk 3]
```diff
@@ -146,6 +151,55 @@ func GetLinkWithPassword(c *gin.Context) {
...
 
+func UpdateLink(c *gin.Context) {
+	var input UpdateLinkInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.JSON(http.StatusUnprocessableEntity, err.Error())
+		return
+	}
+
+	if input.Password == "" && input.Alias == "" {
+		c.JSON(http.StatusNoContent, http.StatusText(http.StatusNoContent))
```
Comment (namtx) [inline controllers/link.go]: 422 it should be more suitable  I think

**Review by namtx — APPROVED** [review]

**Pull request merged**